### DISCUSSION TITLE
Notebook to Webpage

### DISCUSSION THREAD

**Discussion by kovasap** [nextjournal/clerk#225] [Ideas]
One situation I somewhat commonly run into with notebooks goes like this:

1. Mess with a notebook to turn some data file into a visualization or new analyzed data.
2. Package up the notebook into a script or webpage that allows others to re-run my code on new data files of the same shape without my help.

Step (2) can often be a good amount of work.  With clerk, it seems like we could simply create a standalone web app that takes in data files via forms and executes the notebook code on that data as cljs all in the user's browser.  This seems possible to make happen with one clj/lein command that generates this web app (as a single .html file).  

Has anything like this been tried with clerk?  Or do others have good solutions to this problem ?

**Comment by zachcp** [comment]
You could try this repo: https://github.com/zachcp/ghclerk

```
neil new  zachcp/ghclerk ghclerktest \
    --url https://github.com/zachcp/ghclerk \
    --latest-sha

```


If you use this on Github, GH actions will build the image for you. Example [here](https://zachcp.github.io/ghclerktest/#/src/ghclerktest/ghclerktest.clj)

**Comment by kovasap** [comment]
Hmm, unless i'm missing it, that repo does not let you re-execute the notebook code as CLJS on new, user-supplied input.  This is the core of this issue in my eyes.

Thanks for the link though!

**Comment by mk** [comment]
Hey, sorry this fell through the cracks when you originally posted it.

What you can do today is write a notebook so it can be parametrised for example by setting the parameters into an an atom using `v/clerk-eval`. The [Tap Inspector](https://github.com/nextjournal/clerk/blob/79c0dc335022caf63168746c532526ee180b91fa/notebooks/tap.clj) in the clerk repo is an example of this.

This will then only be single-user though and you'd need to provide something like a per-session JVM to make it multi-user. Making this easy is something on our mind but it hasn't taken concrete shape, yet.

Since this issue isn't really actionable, I'm converting it to a discussion.
